pandas/core/ops.py

```diff
@@ -747,28 +753,37 @@ def wrapper(left, right, name=name, na_op=na_op):
         na_op = converted.na_op
 
         if isinstance(rvalues, ABCSeries):
-            name = _maybe_match_name(left, rvalues)
             lvalues = getattr(lvalues, 'values', lvalues)
             rvalues = getattr(rvalues, 'values', rvalues)
             # _Op aligns left and right
         else:
-            name = left.name
             if (hasattr(lvalues, 'values') and
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
+        res_name = _get_series_result_name(left, right)
+
         result = wrap_results(safe_na_op(lvalues, rvalues))
         return construct_result(
             left,
             result,
             index=left.index,
-            name=name,
+            name=res_name,
             dtype=dtype,
         )
 
     return wrapper
 
 
+def _get_series_result_name(left, right):
```

Comment:
rather than right new code, this should be threaded into the existing routine

Reply:
OK. Threading it in separates name-convention logic into multiple places and makes the existing kludge kludgier, but will change.